x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts

```diff
@@ -113,6 +114,22 @@ async function sendEmailWithExchange(
     Authorization: accessToken,
   };
 
+  const axiosInstance = axios.create();
+  axiosInstance.interceptors.response.use(
```

Comment:
Added response interceptor to check for `4xx` errors and delete any stored access tokens for this connector. Previously, if using an access token would result in a `4xx` error, we would still continue to use it until it expired.

Reply:
Tested this by:
* Modifying `x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts` to print out the access token that's used (added console log on line 108)
* Creating an MS Exchange email connector (ask for OAuth credentials if needed) with `from: responseops@elastictest267.onmicrosoft.com`
* Testing it using test connector tab. See the access token being logged and see that a `connector_token` SO is created
* Perform another test and see that the same access token is used as before
* Update the connector to change the from address to ying@elastictest267.onmicrosoft.com. You will have to reenter the client secret
* Perform another test using test connector tab. See error returned and verify that no more `connector_token` SO exists for this connector ID